planner/core/expression_rewriter.go

```diff
@@ -1276,3 +1280,47 @@ func (er *expressionRewriter) toColumn(v *ast.ColumnName) {
 	}
 	er.err = ErrUnknownColumn.GenWithStackByArgs(v.String(), clauseMsg[er.b.curClause])
 }
+
+func (er *expressionRewriter) evalDefaultExpr(v *ast.DefaultExpr) {
+	col, err := er.schema.FindColumn(v.Name)
+	if err != nil {
+		er.err = errors.Trace(err)
+		return
+	}
+
+	table, err := er.b.is.TableByName(col.DBName, col.OrigTblName)
+	if err != nil {
+		er.err = errors.Trace(err)
+		return
+	}
+
+	var val *expression.Constant
+	for _, col := range table.Cols() {
+		if col.Name.L == v.Name.Name.L {
+			// if column default value is 'current_timestamp', use NULL to be compatible with MySQL 5.7
+			if hasCurrentTimestampDefault(col) {
+				val = expression.Null
+			} else {
+				val, er.err = er.b.getDefaultValue(col)
+			}
+			break
+		}
+	}
+	if er.err != nil {
+		return
+	}
+	stkLen := len(er.ctxStack)
+	er.ctxStack = er.ctxStack[:stkLen-1]
+	er.ctxStack = append(er.ctxStack, val)
+}
+
+func hasCurrentTimestampDefault(col *table.Column) bool {
+	if col.Tp != mysql.TypeTimestamp && col.Tp != mysql.TypeDatetime {
```

Comment:
We need to check the types(for example, to avoid `varchar` column with '`current_timestamp`' default string).
What's more, I just found `timestamp` and `datetime` behaves differently here, so a code update is needed, thanks anyway!